python/python/bystro/supervised_ppca/gf_dropout_pt.py

```diff
@@ -0,0 +1,299 @@
+"""
+This implements Factor analysis but with variational inference to allow us
+to supervise a subset of latent variables to be predictive of an auxiliary 
+outcome. Currently only implements logistic regression but in the future 
+will be modified to allow for more general predictive outcomes.
+
+Objects
+-------
+PPCADropout(PPCA)
+    This is PPCA with SGD but supervises a single factor to predict y.
+
+TODO
+SPPCADropout(SPCA)
+    This is SPCA with SGD but supervises a single factor. 
+
+TODO
+FactorAnalysisDropout(FactorAnalysis)
+    This is factor analysis but supervises a single factor.
+
+Methods
+-------
+_get_projection_matrix(W_,sigma_,n_components)
+    Computes the parameters for p(S|X)
+
+"""
+import numpy as np
+
+from tqdm import trange
+import torch
+from torch import nn
+from torch.distributions.multivariate_normal import MultivariateNormal
+from sklearn.decomposition import PCA # type: ignore
+
+from bystro.supervised_ppca._misc_np import softplus_inverse_np
+from bystro.supervised_ppca.gf_generative_pt import PPCA
+
+
+def _get_projection_matrix(W_, sigma_):
+    """
+    This is currently just implemented for PPCA due to nicer formula. Will
+    modify for broader application later.
+
+    Computes the parameters for p(S|X)
+
+    Parameters
+    ----------
+    W_ : pt.Tensor(n_components,p)
+        The loadings
+
+    sigma_ : pt.Flaot
+        Isotropic noise
+
+    Returns
+    -------
+    Proj_X : pt.tensor(n_components,p)
+        Beta such that Proj_XX = E[S|X]
```

Comment:
As written, it looks like a single variable `Proj_XX`.

In general, questions that come up during code review are likely to come up again when other engineers read this code-- could we please default to answering questions *through the code* so that future engineers get the benefit of these clarifications as well?  It's very unlikely that anyone will look back at this code review when reading this code, but if we just amend `Proj_XX` to `Proj_X @ X` (or similar) they won't have to :)